src/mlpack/methods/ann/layer/celu_impl.hpp

```diff
@@ -0,0 +1,62 @@
+/**
+ * @file celu_impl.hpp
+ * @author Gaurav Singh
+ *
+ * Implementation of the CELU activation function as described by Jonathan T. Barron.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_CELU_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_CELU_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "celu.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+CELU<InputDataType, OutputDataType>::CELU(const double alpha) :
+    alpha(alpha),
+    deterministic(false)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename InputType, typename OutputType>
+void CELU<InputDataType, OutputDataType>::Forward(
+    const InputType&& input, OutputType&& output)
+{
+  Fn(input, output);
```

Comment:
There was some discussion about putting the implementation into the `Forward` function itself: https://github.com/mlpack/mlpack/pull/2174/files#r384141238 I guess that applies to here as well?

Reply:
But @zoq, but if the style of this implementation were to be changed by putting `Fn` in `Forward()`, shouldn't the other layers be adjusted as well, to be more consistent? If we are going to do this maybe I can adjust the other implementations to meet the consistency in the next commit.

Reply:
@zoq, @gaurav-singh1998, I will suggest the same in all open PRs for activation functions. I will also open another PR just to make code consistent if it's okay with you. @gaurav-singh1998, I think this will reduce your work for CheckDerivative because if you go with an implementation similar to what is done in activation function, you would have to re-calculate values for all function implemented.

Reply:
Okay, @kartikdutt18 sounds fine.